### ISSUE TITLE
unknown_types false positive: type defined with TYPES BEGIN OF MESH

### ISSUE THREAD

**Issue by rubber-duck-53** [abaplint/abaplint#4150]
```abap
PROGRAM ztest_mesh.

TYPES: BEGIN OF ty_row,
         id TYPE i,
       END OF ty_row.

TYPES: tt_rows TYPE STANDARD TABLE OF ty_row WITH KEY id.

TYPES: BEGIN OF MESH ty_dataset,
         rows TYPE tt_rows
              ASSOCIATION a TO rows
              ON id = id,
       END OF MESH ty_dataset.

DATA dataset TYPE ty_dataset.
```

## Expected

No findings — `ty_dataset` is defined in scope, so `DATA dataset TYPE ty_dataset` resolves.

## Actual

```
ztest_mesh.prog.abap[15, 6] - Variable "DATASET" contains unknown: TY_DATASET not found, lookup (unknown_types) [E]
```

**Comment by larshp** [comment]
@rubber-duck-53 try again, latest version

**Comment by rubber-duck-53** [comment]
Thanks, works as expected now.
